core/src/test/scala/tech/beshu/ror/unit/acl/factory/decoders/rules/auth/JwtAuthorizationRuleSettingsTests.scala

```diff
@@ -0,0 +1,253 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.unit.acl.factory.decoders.rules.auth
+
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.matchers.should.Matchers.*
+import tech.beshu.ror.accesscontrol.blocks.definitions.JwtDef
+import tech.beshu.ror.accesscontrol.blocks.definitions.JwtDef.SignatureCheckMethod
+import tech.beshu.ror.accesscontrol.blocks.rules.auth.JwtAuthorizationRule
+import tech.beshu.ror.accesscontrol.domain.*
+import tech.beshu.ror.accesscontrol.domain.GroupIdLike.GroupId
+import tech.beshu.ror.accesscontrol.factory.RawRorConfigBasedCoreFactory.CoreCreationError.Reason.{MalformedValue, Message}
+import tech.beshu.ror.accesscontrol.factory.RawRorConfigBasedCoreFactory.CoreCreationError.RulesLevelCreationError
+import tech.beshu.ror.providers.EnvVarProvider.EnvVarName
+import tech.beshu.ror.providers.EnvVarsProvider
+import tech.beshu.ror.unit.acl.factory.decoders.rules.BaseRuleSettingsDecoderTest
+import tech.beshu.ror.utils.TestsUtils.*
+import tech.beshu.ror.utils.uniquelist.UniqueNonEmptyList
+
+import java.security.KeyPairGenerator
+import java.util.Base64
+
+class JwtAuthorizationRuleSettingsTests
```

Comment:
It looks like it should have similar tests to the RorKbnAuthorizationRuleSettingsTests suite. Shouldn't it?

Reply:
I added 3 additional test cases similar to ROR KBN
```
      "no JWT definition name is defined"
      "both 'groups or' key and 'groups and'"
      "two JWT definitions have the same names"
```

The others are just concerning token format issues, and tests for that are already tested in `JwtAuthRuleSettingsTests` (tests specific to JWT definition, and not the rule itself)